### PR TITLE
Minimap and powerup icon settings in user settings

### PR DESCRIPTION
**Pull request by luffah**
Hi,
i propose this PR in order to allow to customize the dashboard shown while racing
by editing local user config (`config.xml`).

The PR does (with motives):
 - add size and display settings (center, right, hidden) for powerup icon  (motives below)
     - powerup icon may strongly obstruct field of view with custom camera settings
     - powerup icon doesn't provide any additional information if the option to show karts help powerup is active
     - in Soccer Tournament  context, some players wish to remove the powerup icon in order to record the match as spectator (other gui elements are not obstructing but may be subject for future PR)
 - move minimap display settings in user settings (motives below)
      - minimap size is not similar to other core (mostly physics, rules and api) properties define in stk_config
      - minimap size goes well with others ui setting  defined in user settings
      - stk_config is not accessible on some packaged version - e.g. snap
 - fix some comments in stk_config.xml
 
 Limits of the settings :
   - powerup size
       - only sizes between 16 and 64 (16, 24, 32, 48, 64 icon standards) are actually usable values
       - sizes < 32 looks poorly integrated in gui
       - using size 1 or 2 make the icon looks like a deadpixel
   - powerup display
        - the user shall may use  "kart's held powerup" and "soccer player list" settings  if minimap hidden
        - on very small screen the setting indicating "center" position will be overridden pushing the icon to right corner (current  behavior)
   - minimap size
        - ~~users modifying this setting in stk_config.xml  will have to edit local config.xml instead~~ 
   - minimap display
        - users may have to set this setting again after update because both size and display are with this PR attribute of the same xml tag

Best regards.

## Agreement
```
By creating a pull request in stk-code, you hereby agree to dual-license your contribution as
GNU General Public License version 3 or any later version and
Mozilla Public License version 2 or any later version.

This includes your previous contribution(s) under the same name of contributor.

Keep the above statement in the pull request comment for agreement.

```

### CODE REVIEW

File: data/kart_characteristics.xml [hunk 1]
```diff
@@ -145,7 +145,6 @@
             visual-time: How long it takes for the visual skid to reach maximum.
             revert-visual-time: how long it takes when stopping a skid to revert
                 the visual skid and bring visuals and physics in sync again.
-            angular-velocity: Angular velocity to be used for the kart when skidding.
```
Comment (luffah) [inline data/kart_characteristics.xml:148]: angular-velocity was nowhere to be found

File: data/stk_config.xml [hunk 2]
```diff
@@ -281,40 +280,7 @@
        ============================ -->
   <general-kart-defaults>
 
-    <!-- Skidding: increase: multiplicative increase of skidding factor in each frame.
```
Comment (luffah) [inline data/stk_config.xml:284]: This "Skidding" comments is relatade to no xml tag and is exactly the same in another part of the file.

File: data/stk_config.xml [hunk 3]
```diff
@@ -527,11 +490,10 @@
               visual-time: How long it takes for the visual skid to reach maximum.
               revert-visual-time: how long it takes when stopping a skid to revert
                 the visual skid and bring visuals and physics in sync again.
-              angular-velocity: Angular velocity to be used for the kart when skidding.
               min-speed: Minimum speed a kart must have before it can skid. Must be
                 >0, otherwise the kart can skid at the start of the race.
               time-till-bonus: How long a kart needs to skid in order to get a bonus.
-              bonus-force: A speedup applied to the kart whick skidded for a while.
```
Comment (luffah) [inline data/stk_config.xml:534]: property name was not exact in the comment

File: src/config/stk_config.cpp [hunk 4]
```diff
@@ -174,9 +174,6 @@ void STKConfig::load(const std::string &filename)
     CHECK_NEG(m_replay_delta_steering,     "replay delta-steering"      );
     CHECK_NEG(m_replay_delta_speed,        "replay delta-speed     "    );
     CHECK_NEG(m_replay_dt,                 "replay delta-t"             );
```
Comment (luffah) [inline src/config/stk_config.cpp:176]: removing the minimap from stk_config -> this will go in user_config

File: data/stk_config.xml [hunk 5]
```diff
@@ -188,13 +189,7 @@
   <replay max-frames="12000" delta-t="0.100"  delta-speed="0.6"
           delta-steering="0.26" />
```
Comment (luffah) [inline data/stk_config.xml:191]: remove minimap setting from stk_config -> will appear in user_config (minimap tag)

**Comment by NickerAban** [comment]
M

**Comment by Benau** [comment]
You should know that stk prefer as little options as possible.....

**Pull request merged**